src/User/Types.hs

```diff
@@ -1,11 +1,34 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
-{-# LANGUAGE RecordWildCards, TypeSynonymInstances, FlexibleInstances, OverloadedStrings #-}
+{-# LANGUAGE TypeSynonymInstances #-}
 module User.Types where
 
 import Data.Time.Calendar
 import Snap.Snaplet.PostgresqlSimple
 import Database.PostgreSQL.Simple.ToField
 import Database.PostgreSQL.Simple.FromField
+import qualified Data.ByteString.Char8 as B
+import Web.PathPieces
+import qualified Data.Text as T
+import Text.Read
+import Text.Printf
+
+newtype Currency 
+  = Currency 
+  { unCurrency :: Double } 
+  deriving (Show,Eq)
+
+instance ToField Currency where
+  toField = toField . unCurrency
+
+instance FromField Currency where
+  fromField _ md = return $ Currency $
+    case md of
+      Just d -> read $ B.unpack d :: Double
+      _      -> 0
```

Comment:
@kinorius123 I think we should throw an error here. It musn't happen that you get `Nothing` when the table is `NOT NULL`.